### DISCUSSION TITLE
data from private profiles shown in Bionomia but excluded from csv downloads

### DISCUSSION THREAD

**Discussion by abubelinha** [bionomia/bionomia#372] [Q&A]
Hi.   I am trying to make a report **_for_** the people I helped, where I would like to call them by their real names as they are shown in Bionomia (instead of using their IDs).

My plan was inner-joining my atributions.csv download to bionomia-public-profiles.csv , but I didn't recall the "public" in it's name. 
Now I realize I am missing most of the names since private profiles are not available in that file.
But ... why hiding them from .csv and not from .html?

Bionomia web is already showing all that information, as it comes from ORCID/Wikidata. 
https://en.bionomia.net/help-others/0000-0001-6449-7181
Name, surname, nationality ... are already visible in that kind of links. 
I understand it means those people decided to make that info public.
So why not showing the same info in Bionomia .csv files, just like it is being done with public profiles?

Also, a possible bug.  Why are some Wikidata profiles of dead people being shown as private?
https://en.bionomia.net/Q21523046
I found some others like this one.  
But I think their names should be available in csv profiles too.

I hope I am not missing something already explained somewhere.

Thanks a lot in advance
@abubelinha

**Comment by dshorthouse** [comment]
> My plan was inner-joining my atributions.csv download to bionomia-public-profiles.csv , but I didn't recall the "public" in it's name. Now I realize I am missing most of the names since private profiles are not available in that file. But ... why hiding them from .csv and not from .html?

I guess the idea was to merely provide the basics in the csv for a user to gather the metadata from their sources via the provided identifiers. I've just [committed](https://github.com/bionomia/bionomia/commit/acce107a0b4581cf65dbd167cafdb6d9256ab9a3) the addition of the "label". The other metadata are relational & not conducive to the construction of a flat csv download.
 
> Bionomia web is already showing all that information, as it comes from ORCID/Wikidata. https://en.bionomia.net/help-others/0000-0001-6449-7181 Name, surname, nationality ... are already visible in that kind of links. I understand it means those people decided to make that info public. So why not showing the same info in Bionomia .csv files, just like it is being done with public profiles?

The /help-others/ paths are not public; they require an active session via an ORCID pass-thru login.
 
> Also, a possible bug. Why are some Wikidata profiles of dead people being shown as private? https://en.bionomia.net/Q21523046 I found some others like this one. But I think their names should be available in csv profiles too.

In the /help-others/ pages, there's a link to make wikidata-based profiles public in Bionomia. There are tens of thousands of wikidata-based profiles in Bionomia and many have no linked specimen records. A link for scribes to make these public was a simple way to act as a filter.

<img width="518" alt="Screenshot 2024-07-10 at 1 49 51 PM" src="https://github.com/bionomia/bionomia/assets/326459/99556c83-29fd-43ee-879f-b17ec8020759">

**Comment by abubelinha** [comment]
> In the /help-others/ pages, there's a link to make wikidata-based profiles public in Bionomia

Oh! Thanks a lot, I had not noticed that link.  That's enough for what I need.
But for those with linked specimen records, wouldn't it make sense they being public by default?

> The /help-others/ paths are not public; they require an active session via an ORCID pass-thru login.

Sorry. I forgot I was logged in, so I pasted the wrong link. 

I was indeed meaning this kind of link:
https://bionomia.net/0000-0001-6449-7181
(where the ORCID name info is also visible for non-logged people)

That one is an ORCID-based, non-public Bionomia profile.
But the person's name is visible in Bionomia for everybody.
(and I think that is OK, since it is already visible in ORCID, and creating that ORCID was the user wish).

So, why don't let this info go into a new downloadable "bionomia-non-public-profiles.csv"? 
(just in case you wan to keep public/non-public separated in to different files)
Otherwise, just a single file with all Bionomia profiles (public/private in a flag column) which have linked specimen records.

**Reply by dshorthouse** [reply]
Will think about this.

In the meantime, I've introduced a cron job to make wikidata profiles public that have at least one attribution that have not yet been made public.

**Comment by abubelinha** [comment]
Thanks a lot. 
For private profiles I will use https://api.bionomia.net/{person.id}.jsonld in the mean time, no rush. 
But having those ids & names in the downloadable whatever-profiles.csv would simplify it.
